code/codegen/src/main/kotlin/com/android/gradle/replicator/codegen/GenerationParameters.kt

```diff
@@ -0,0 +1,103 @@
+package com.android.gradle.replicator.codegen
+
+import java.io.File
+
+/**
+ * Parameters for the generation
+ */
+data class GenerationParameters(
+        /**
+         * classpath available to import types from.
+         */
+        val classpath: List<File>,
+        /**
+         * Java Language level supported for the generation.
+         */
+        val javaLanguageLevel: String,
+
+        /**
+         * Parameters specific to class generation.
+         */
+        val classGenerationParameters: ClassGenerationParameters
+) {
+    class Builder {
+        private val classpath = mutableListOf<File>()
+        private var javaLanguageLevel: String? = null
+        val classGenerationParametersBuilder = ClassGenerationParameters.Builder()
+
+
+        fun addClasspathElement(element: File) {
+            classpath.add(element)
+        }
+
+        fun setJavaLanguageLevel(languageLevel: String) {
+            this.javaLanguageLevel = languageLevel
+        }
+
+        fun build(): GenerationParameters =
+                GenerationParameters(
+                        classpath = classpath.toList(),
+                        javaLanguageLevel = javaLanguageLevel ?: "1.8",
+                        classGenerationParameters = classGenerationParametersBuilder.build()
+                )
+    }
+}
+
+data class ClassGenerationParameters(
+        // min number of instance variables
+        val minNumberOfInstanceVars: Int,
+        // max number of instance variables
+        val maxNumberOfInstanceVars: Int,
+        // min number of methods to generate
+        val minNumberOfMethods: Int,
+        // max number of methods of generate
+        val maxNumberOfMethods: Int,
+
+        // max number of parameters for method.
+        val maxNumberOfMethodParameters: Int,
+        // min number of code blocks in method declaration.
+        val minNumberOfMethodBlocks: Int,
+        // max number of code blocks in method declaration.
+        val maxNumberOfMethodBlocks: Int,
+
+        // max number of blocks in if() condition
+        val maxNumberOfBlocksInIf: Int,
+        // max number of blocks in ifElse() condition
+        val maxNumberOfBlocksInIfElse: Int,
+
+        // max number of blocks in lambda declaration.
+        val maxNumberOfBlocksInLambda: Int,
+
+        // max number of blocks in loop
+        val maxNumberOfBlocksInLoop: Int
+) {
+    class Builder(
```

Comment:
is this useful? why not put the default value on the data class constructor directly?

Reply:
it makes it easier to test and eventually I will save those into a parameter file that can be edited.